finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Row.scala

```diff
@@ -71,14 +71,15 @@ class StringEncodedRow(rawRow: Buf, val fields: IndexedSeq[Field], indexMap: Map
       else {
         val str = new String(bytes, Charset(charset))
         field.fieldType match {
-          case Type.Tiny       => ByteValue(str.toByte)
-          case Type.Short      => ShortValue(str.toShort)
-          case Type.Int24      => IntValue(str.toInt)
-          case Type.Long       => IntValue(str.toInt)
-          case Type.LongLong   => LongValue(str.toLong)
-          case Type.Float      => FloatValue(str.toFloat)
-          case Type.Double     => DoubleValue(str.toDouble)
-          case Type.Year       => ShortValue(str.toShort)
+          case Type.Tiny                        => ByteValue(str.toByte)
+          case Type.Short                       => ShortValue(str.toShort)
+          case Type.Int24                       => IntValue(str.toInt)
+          case Type.Long if field.isUnsigned()  => LongValue(str.toLong)
```

Comment:
Yeah and if the data type of the column in MySQL, is unsigned long then the type of the field in finagle side is Type.LongLong. 

BTW by saying 
> The column should be able to be unsigned long
You mean BIGINT UNSIGNED, right. Is there a data type long n MySQL?